pkg/controller/seed-controller-manager/addon/addon_controller.go

```diff
@@ -533,6 +534,62 @@ func (r *Reconciler) migrateHetznerCSIDriver(ctx context.Context, log *zap.Sugar
 	return nil
 }
 
+func (r *Reconciler) unInstallCSIDriver(ctx context.Context, log *zap.SugaredLogger, addon *kubermaticv1.Addon, cluster *kubermaticv1.Cluster) error {
+	userClusterClient, err := r.KubeconfigProvider.GetClient(ctx, cluster)
+	if err != nil {
+		return fmt.Errorf("failed to get client for usercluster: %w", err)
+	}
+	// Increased the timeout, some of the operations were getting timed out.
+	ctx, cancel := context.WithTimeout(context.Background(), 60*time.Second)
```

Comment:
Please don't introduce new contexts and use the one already propagated.

```suggestion
	ctx, cancel := context.WithTimeout(ctx, 60*time.Second)
```